### PR TITLE
Use for-await in examples and manual

### PR DESCRIPTION
**Pull request by hayd**
follow up to #3212, see also #3216.

There's still a top-level for-await in the website home page, but this should be fixed after 0.22 (once this syntax is released).

https://github.com/denoland/deno_website2/blob/92a20e81c51c051080311c2edefe517049dbafcd/src/Home.js#L7-L15

### CODE REVIEW

File: std/examples/curl.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
 // Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
-const url = Deno.args[1];
-const res = await fetch(url);
+const url_ = Deno.args[1];
+const res = await fetch(url_);
```
Comment (ry) [inline std/examples/curl.ts]: Why this change?

File: std/http/http_bench.ts [hunk 2]
```diff
@@ -5,11 +5,7 @@ const addr = Deno.args[1] || "127.0.0.1:4500";
...
 
-async function main(): Promise<void> {
-  console.log(`http://${addr}/`);
-  for await (const req of server) {
-    req.respond({ body });
-  }
+console.log(`http://${addr}/`);
+for await (const req of server) {
+  req.respond({ body });
 }
```
Comment (ry) [inline std/http/http_bench.ts]: Much nicer.

File: std/examples/curl.ts (hunk 1 above)
Comment (hayd) [reply std/examples/curl.ts]: Ah. This is currently an error

Comment (hayd) [reply std/examples/curl.ts]: Which is to say, with url it didn't compile. #2888 

This was one file I didn't actually verify (thought it was a sure thing).

**Comment by hayd** [comment]
@ry I pushed updates for tools/deno_tcp.ts. Also made PR for the website: https://github.com/denoland/deno_website2/pull/87
